<launch>

  <arg name="initial_file" default="$(find bwi_planning)/experiment/initial" />

  <arg name="domain_costs_file" default="$(find bwi_planning)/domain/cost" />
  <arg name="domain_semantics_file" default="$(find bwi_planning)/domain/nav" />

  <arg name="query_file" default="$(find bwi_planning)/experiment/query" />
  <arg name="costs_file" default="$(find bwi_planning)/experiment/distances.lua" />
  <arg name="rigid_knowledge_file" default="$(find bwi_planning)/domain/experiment" />

  <arg name="planning_times_file" default="/tmp/planning_times.txt" />

  <arg name="dry_run" default="false" />
  <arg name="auto_open_door" default="true" />
  <arg name="use_cost_optimization" default="false" />
  <arg name="enable_learning" default="false" />

  <node name="planner_krr2014" type="planner_krr2014" pkg="bwi_planning"
    output="screen" required="true">

    <param if="$(arg dry_run)" name="initial_file" value="$(arg initial_file)" />

    <param if="$(arg use_cost_optimization)" 
      name="costs_file" value="$(arg costs_file)" />
    <param if="$(arg use_cost_optimization)" 
      name="domain_costs_file" value="$(arg domain_costs_file)" />

    <param name="query_file" value="$(arg query_file)" />
    <param name="domain_semantics_file" value="$(arg domain_semantics_file)" />
    <param name="rigid_knowledge_file" value="$(arg rigid_knowledge_file)" />

    <param name="dry_run" value="$(arg dry_run)" />
    <param name="auto_open_door" value="$(arg auto_open_door)" />

    <param name="enable_learning" value="$(arg enable_learning)" />
    <param if="$(arg enable_learning)" name="planning_times_file"
      value="$(arg planning_times_file)" />
    
  </node>

</launch>
